tests/spin/test_lattice.py

```diff
@@ -0,0 +1,518 @@
+                [1.0, 1.15470054],
+                [1.5, 1.44337567],
+                [1.5, 0.28867513],
+                [2.0, 0.57735027],
+                [2.0, 1.15470054],
+                [2.5, 1.44337567],
+            ],
+        ),
+    ],
+)
+def test_positions(vectors, positions, n_cells, expected_points):
+    r"""Test that the lattice points start from the coordinates provided in the positions"""
+
+    lattice = Lattice(n_cells=n_cells, vectors=vectors, positions=positions)
+    assert np.allclose(expected_points, lattice.lattice_points)
+
+
+@pytest.mark.parametrize(
+    ("vectors", "positions", "n_cells", "expected_number"),
+    # expected_number here was obtained manually.
+    [
+        ([[0, 1], [1, 0]], [[0, 0]], [3, 3], 9),
+        ([[0, 1], [1, 0]], [[0, 0]], [6, 7], 42),
+        ([[1, 0], [0.5, np.sqrt(3) / 2]], [[0.5, 0.5 / 3**0.5], [1, 1 / 3**0.5]], [2, 2], 8),
+        (np.eye(3), None, [3, 3, 4], 36),
+    ],
+)
+def test_lattice_points(vectors, positions, n_cells, expected_number):
+    r"""Test that the correct number of lattice points are generated for the given attributes"""
+    lattice = Lattice(n_cells=n_cells, vectors=vectors, positions=positions)
+    assert len(lattice.lattice_points == expected_number)
+
+
+@pytest.mark.parametrize(
+    # expected_edges here were obtained manually.
+    ("vectors", "positions", "n_cells", "boundary_condition", "expected_edges"),
+    [
+        (
+            [[0, 1], [1, 0]],
+            [[0, 0]],
+            [3, 3],
+            [True, True],
+            [
+                (0, 1, 0),
+                (1, 2, 0),
+                (3, 4, 0),
+                (5, 8, 0),
+                (6, 8, 0),
+                (0, 3, 0),
+                (1, 4, 0),
+                (0, 6, 0),
+                (4, 7, 0),
+                (6, 7, 0),
+                (1, 7, 0),
+                (0, 2, 0),
+                (4, 5, 0),
+                (3, 6, 0),
+                (2, 5, 0),
+                (3, 5, 0),
+                (7, 8, 0),
+                (2, 8, 0),
+            ],
+        ),
+        (
+            [[0, 1], [1, 0]],
+            [[0, 0]],
+            [3, 4],
+            [True, False],
+            [
+                (3, 7, 0),
+                (8, 9, 0),
+                (0, 8, 0),
+                (1, 9, 0),
+                (4, 5, 0),
+                (5, 6, 0),
+                (4, 8, 0),
+                (5, 9, 0),
+                (9, 10, 0),
+                (0, 1, 0),
+                (10, 11, 0),
+                (0, 4, 0),
+                (1, 2, 0),
+                (1, 5, 0),
+                (2, 10, 0),
+                (6, 7, 0),
+                (6, 10, 0),
+                (3, 11, 0),
+                (2, 3, 0),
+                (2, 6, 0),
+                (7, 11, 0),
+            ],
+        ),
+        (
+            [[1, 0], [0.5, np.sqrt(3) / 2]],
+            [[0.5, 0.5 / 3**0.5], [1, 1 / 3**0.5]],
+            [2, 2],
+            True,
```

Comment:
No boundary conditions to test here?

Reply:
Not sure I follow, there is True to represent periodic boundary conditions in all directions of the grid.

Reply:
Will add another test for that.